src/video/kmsdrm/SDL_kmsdrmvideo.c

```diff
@@ -278,8 +317,7 @@ int KMSDRM_VideoInit(_THIS, SDL_PixelFormat *vformat)
 
 SDL_Rect **KMSDRM_ListModes(_THIS, SDL_PixelFormat *format, Uint32 flags)
 {
-	/** TODO:: Unstub. **/
-   	return (SDL_Rect **) -1;
+	return drm_vid_modes;
```

Comment:
For the user to be able to set arbitrary, non-native resolutions and let DRM sort it out, it's necessary to hack around SDL_SetVideoMode to attempt setting modes that are unlisted here.

This is necessary for applications that won't respect -1 being "any mode is supported".